openedx/core/djangoapps/courseware_api/views.py

```diff
@@ -815,9 +815,11 @@ def get(self, request, usage_key_string, *args, **kwargs):  # lint-amnesty, pyli
                 view = STUDENT_VIEW
                 if request.user.is_anonymous:
                     view = PUBLIC_VIEW
-
+                enrollment = CourseEnrollment.get_enrollment(request.user, usage_key.course_key)
```

Comment:
I have the same comment here as above. I'd recommend you use [CourseEnrollment.enrollment_mode_for_user](https://github.com/edx/edx-platform/blob/c523a5d9dde32f48c3b40e2eacfc652a7935ee25/common/djangoapps/student/models/course_enrollment.py#L941) to get the learner's enrollment mode